koku/api/settings/test/tag_mappings/test_view.py

```diff
@@ -0,0 +1,288 @@
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_203_NON_AUTHORITATIVE_INFORMATION:
+            # Removing children
+            url = reverse("tags-mapping-child-remove")
+            data = {"ids": [random_uuid_list[1], random_uuid_list[3]]}
+            response = self.client.put(url, data, format="json", **self.headers)
+
+            self.assertEqual(response.status_code, status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+
+    def test_put_method_remove_parent(self):
+        """Test removing parent."""
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_203_NON_AUTHORITATIVE_INFORMATION:
+            # Removing parent
+            url = reverse("tags-mapping-parent-remove")
+            data = {"ids": [random_uuid_list[0]]}
+            response = self.client.put(url, data, format="json", **self.headers)
+            self.assertEqual(response.status_code, status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+            self.assertEqual(response.data["detail"], "Parents deleted successfully.")
+
+    def retrieve_sample_uuids(self):
+        """Gets all inserted uuids to use on adding(put) methods."""
+        with tenant_context(self.tenant):
+            enabled_tag_keys = EnabledTagKeys.objects.all()
+            uuid_list = [tag_key.uuid for tag_key in enabled_tag_keys]
+
+        return uuid_list
+
+    def test_filter_by_source_type(self):
+        """Test the filter by source_type."""
+
+        filter = SettingsTagMappingFilter()
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        # Get an already inserted provider type to check if the filter is working
+        parent_provider_types = TagMapping.objects.values_list("parent__provider_type", flat=True).distinct()
+        test_filter = parent_provider_types[0]
+
+        # Call the filter_by_source_type method with 'test_filter' as the value
+        if response.status_code == status.HTTP_203_NON_AUTHORITATIVE_INFORMATION:
```

Comment:
This test should be updated to remove this if statement. The above should be written so that the response is always the correct status code which ensures the code below is run.